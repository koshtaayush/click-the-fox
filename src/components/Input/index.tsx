//Library
import React from 'react'
import styled from 'styled-components'

interface Props {
    onChangeProp: (val: string) => void
    inputValue: string
    placeHolderText: string
}

const Input: React.FC<Props> = (props) => {
    const { onChangeProp, inputValue, placeHolderText } = props

    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const { value } = e.target
        if (onChangeProp) {
            onChangeProp(value)
        }
    }

    return (
        <React.Fragment>
            <InputBox
                type="text"
                placeholder={placeHolderText}
                onChange={handleInputChange}
                value={inputValue}
                test-id="input"
            />
        </React.Fragment>
    )
}

export default Input

const InputBox = styled.input`
    border: 0.125rem solid #FFC000;
    color: #000000;
    border-radius: 0.25rem;
    padding: 0.5rem;
    outline: none;
`
