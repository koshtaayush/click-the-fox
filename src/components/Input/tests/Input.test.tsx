import React from 'react'
import { shallow } from 'enzyme'

import Input from '..'

describe('Input', () => {
    it('Should match the snapshot', () => {
        const mockFn = jest.fn()

        const component = shallow(
            <Input
                placeHolderText={'Dummy placeholder text'}
                onChangeProp={mockFn}
                inputValue={'Hello'}
            />
        )

        expect(component).toMatchSnapshot()
    })

    it('Should render the input component correctly', () => {
        const mockFn = jest.fn()

        const component = shallow(
            <Input
                placeHolderText={'Dummy placeholder text'}
                onChangeProp={mockFn}
                inputValue={'Hello'}
            />
        )

        expect(component.find({ 'test-id': 'input' })).toHaveLength(1)
    })

    it('Should pass the same value which is passed', () => {
        const mockFn = jest.fn()

        const component = shallow(
            <Input
                placeHolderText={'Dummy placeholder text'}
                onChangeProp={mockFn}
                inputValue={'Hello'}
            />
        )

        expect(component.find({ 'test-id': 'input' }).props().value).toEqual(
            'Hello'
        )
    })
})
