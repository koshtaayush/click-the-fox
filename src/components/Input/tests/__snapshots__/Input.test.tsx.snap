// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should match the snapshot 1`] = `
<Fragment>
  <styled.input
    onChange={[Function]}
    placeholder="Dummy placeholder text"
    test-id="input"
    type="text"
    value="Hello"
  />
</Fragment>
`;
